MapboxMobileEvents/MMEMetricsManager.m

```diff
@@ -0,0 +1,160 @@
+#import "MMEMetricsManager.h"
+#import "MMEEvent.h"
+#import "MMEReachability.h"
+#import "MMEConstants.h"
+
+@interface MMEMetricsManager ()
+
+@property (nonatomic) int requests;
+@property (nonatomic) long totalDataTransfer;
+@property (nonatomic) long cellDataTransfer;
+@property (nonatomic) long wifiDataTransfer;
+@property (nonatomic) int appWakeups;
+@property (nonatomic) int eventCountFailed;
+@property (nonatomic) int eventCountTotal;
+@property (nonatomic) int eventCountMax;
+@property (nonatomic) int deviceTimeDrift;
+@property (nonatomic) float deviceLat;
+@property (nonatomic) float deviceLon;
+@property (nonatomic) NSDate *dateUTC;
+@property (nonatomic) NSString *dateUTCString;
+@property (nonatomic) NSDictionary *configResponseDict;
+@property (nonatomic) NSMutableDictionary *eventCountPerType;
+@property (nonatomic) NSMutableDictionary *failedRequestsDict;
+
+@end
+
+@implementation MMEMetricsManager
+
++ (instancetype)sharedManager {
+    static MMEMetricsManager *_sharedManager;
+    static dispatch_once_t onceToken;
+    
+    dispatch_once(&onceToken, ^{
+        _sharedManager = [[MMEMetricsManager alloc] init];
+    });
+    
+    return _sharedManager;
+}
+
+- (void)metricsFromEventQueue:(NSArray *)eventQueue {
+    if (self.dateUTC == nil) {
+        [self updateDateUTC];
+    }
+    
+    if (eventQueue.count > 0) {
+        if (self.eventCountPerType == nil) {
+            self.eventCountPerType = [[NSMutableDictionary alloc] init];
+        }
+        
+        self.eventCountTotal = self.eventCountTotal + (int)eventQueue.count;
+        
+        for (MMEEvent *event in eventQueue) {
+            NSNumber *eventCount = [self.eventCountPerType objectForKey:event.name];
+            eventCount = [NSNumber numberWithInteger:[eventCount integerValue] + 1];
+            [self.eventCountPerType setObject:eventCount forKey:event.name];
+        }
+    }
+}
+
+- (void)metricsFromEvents:(nullable NSArray *)events error:(nullable NSError *)error {
+    if (error == nil) {
```

Comment:
```suggestion
    if (error == nil) {
        self.requests += 1;
        return;
    }
    // content of else block...
    
```
Btw, what's the guideline around `self.requests += 1;` vs `self.requests++;`? are both allowed/idiomatic?

Reply:
Ahh, you're right -- I had thought this was decremented on ObjC as well 🤦‍♂️ (+= would have been better even so). Thanks!